src/libNode/Node.cpp

```diff
@@ -1923,6 +1937,13 @@ bool Node::ProcessTxnPacketFromLookupCore(const bytes& message,
     P2PComm::GetInstance().SendBroadcastMessage(toSend, message);
   }
 
+  if (m_mediator.m_ds->m_mode == DirectoryService::Mode::IDLE &&
+      m_state != MICROBLOCK_CONSENSUS_PREP) {
+    unique_lock<mutex> lk(m_mutexCVWaitDSBlock);
+    cv_txnPacket.wait(lk,
+                      [this] { return m_state == MICROBLOCK_CONSENSUS_PREP; });
+  }
+
```

Comment:
by the time cv_txnPacket is notified ( after moving to MICROBLOCK_CONSENSUS_PREP), its seems possible that leader/backup already start processing transaction.
Here i think we assume all pkts are checked, validated and added to m_createdTxns by then which may not hold true.

lock_guard<mutex> g(m_mutexCreatedTransactions) at Line 2026 will give sequential processing. so with multiple packets, it may fail to check and add to mem pool before , leader actually start processing at Line 1100

Reply:
i see there is sleep for `TXN_DISTRIBUTION_TIME_MS` but still there is chance for race condition provided it can be as small as 1 sec.
May be need some cv to indicate all pkts were checked and added to pool before starting processing txns. ofcorse need to some timeout for that as well.
Basically give enough time to do validation and adding to mempool for all pkts that were received.

Reply:
yup very good point, will address it